x-pack/plugins/ml/public/application/util/chart_utils.test.js

```diff
@@ -33,242 +33,484 @@ import { mount } from 'enzyme';
-    const wrapper = mount(<div className="content-wrapper" />);
-    const node = wrapper.getDOMNode();
-
-    const chartHeight = 170;
-    const margin = { top: 10, right: 0, bottom: 30, left: 60 };
-    const svgWidth = 500;
-    const svgHeight = chartHeight + margin.top + margin.bottom;
-    const vizWidth = 500;
-
-    const chartElement = d3.select(node);
-
-    const lineChartXScale = d3.time
-      .scale()
-      .range([0, vizWidth])
-      .domain([plotEarliest, plotLatest]);
-
-    const xAxis = d3.svg
-      .axis()
-      .scale(lineChartXScale)
-      .orient('bottom')
-      .innerTickSize(-chartHeight)
-      .outerTickSize(0)
-      .tickPadding(10)
-      .tickFormat(d => moment(d).format(xAxisTickFormat));
-
-    const tickValues = getTickValues(startTimeMs, interval, plotEarliest, plotLatest);
-    xAxis.tickValues(tickValues);
-
-    const svg = chartElement
-      .append('svg')
-      .attr('width', svgWidth)
-      .attr('height', svgHeight);
-
-    const axes = svg.append('g');
-
-    const gAxis = axes
-      .append('g')
-      .attr('class', 'x axis')
-      .attr('transform', 'translate(0,' + chartHeight + ')')
-      .call(xAxis);
-
-    return {
-      gAxis,
-      node,
-      vizWidth,
-    };
-  }
+  describe('getXTransform', () => {
+    const expectedXTransform = 0.007167499999999999;
 
-  test('farequote sample data', () => {
-    const mockedGetBBox = { width: 27.21875 };
-    SVGElement.prototype.getBBox = () => mockedGetBBox;
+    test('Chrome/Safari/Firefox String variant.', () => {
+      const transformStr = 'translate(0.007167499999999999,0)';
+      const xTransform = getXTransform(transformStr);
+      expect(xTransform).toEqual(expectedXTransform);
+    });
 
-    const startTimeMs = 1486656000000;
-    const interval = 14400000;
+    test('IE11 String variant.', () => {
+      const transformStr = 'translate(0.007167499999999999)';
+      const xTransform = getXTransform(transformStr);
+      expect(xTransform).toEqual(expectedXTransform);
+    });
 
-    const { gAxis, node, vizWidth } = axisSetup({
-      interval,
-      plotEarliest: 1486606500000,
-      plotLatest: 1486719900000,
-      startTimeMs,
-      xAxisTickFormat: 'HH:mm',
+    test('Invalid String.', () => {
+      const transformStr = 'translate()';
+      const xTransform = getXTransform(transformStr);
+      expect(xTransform).toEqual(NaN);
     });
+  });
 
-    expect(node.getElementsByTagName('text')).toHaveLength(8);
+  describe('removeLabelOverlap', () => {
+    const originalGetBBox = SVGElement.prototype.getBBox;
+
+    // This resembles how ExplorerChart renders its x axis.
+    // We set up this boilerplate so we can then run removeLabelOverlap()
+    // on some "real" structure.
+    function axisSetup({ interval, plotEarliest, plotLatest, startTimeMs, xAxisTickFormat }) {
+      const wrapper = mount(<div className="content-wrapper" />);
```

Comment:
nit: in the most recent tests we use `@testing-library/react` `render` instead of `enzyme` `mount`, might be worth to update here as well

Reply:
Done in https://github.com/elastic/kibana/pull/63132/commits/09cb0ec9d106422c07d7a72ebaaf7288862d76ea